### ISSUE TITLE
IsPersistent needs a overload for Type

### ISSUE THREAD

**Issue by malx122** [Starcounter/Home#427]
If I have an object I can check if it is of a Persistent (database) type with the IsPersistent() extension.
But if I just have a System.Type (no instance of it) I can not find a way of checking if its of a database type.

I suggest that you add an overload extension for System.Type so IsPersistent() returns whether the type is persistent instead of checking the actual Type object (that of course always returns false)

**Comment by miyconst** [comment]
@per-samuelsson maybe we already have something like this?

**Comment by per-samuelsson** [comment]
> @per-samuelsson maybe we already have something like this?

Nothing packaged in a nice API, but there are several ways to achieve this. One is by checking if there is a type definition by the full name:

```csharp
Bindings.GetTypeDef(type.FullName)
```

That returns `null` for types not bound (i.e. non-database types).

Another, probably more "correct", way would be to query metadata, but I don't remember exactly what table to query.

@malx122,

What about you create that extension method in your code-base, until we reach Nova and see what we want to surface there? Like

```csharp
static class TypeExtensions {
  public static bool IsPersistent(this Type type) {
    return Bindings.GetTypeDef(type.FullName) != null;
  }
}
```

**Comment by miyconst** [comment]
>Another, probably more "correct", way would be to query metadata, but I don't remember exactly what table to query.

It is possible to check with the following SQL query:

```sql
SELECT COUNT(*)
FROM Starcounter.Metadata.RawView
WHERE FullName = 'Starcounter.Metadata.RawView'
```

**Comment by malx122** [comment]
Ok, thanks for the tips. Changed my "not good but worked for the current use" extension, `type.IsClass && !type.FullName.Contains("System")` to the one suggested above.
Still think it should be part of the future SC!

**Comment by miyconst** [comment]
Closing as won't fix in Starcounter 2.4 and not applicable in Starcounter 3.0.
